docs/user-guide/petset.md

```diff
@@ -0,0 +1,269 @@
+* [Headless Services](/docs/user-guide/gc/services/#headless-services)
+* [Persistent Volumes](/docs/user-guide/gc/volumes.md)
+* [Dynamic volume provisioning](http://releases.k8s.io/{{page.githubbranch}}/examples/experimental/persistent-volume-provisioning/README.md)
+
+You need a working Kubernetes cluster at version >= 1.3, with a healthy DNS [cluster addon](http://releases.k8s.io/{{page.githubbranch}}/cluster/addons/README.md) at version >= 15. You cannot use Pet Set on a hosted Kubernetes provider that has disabled `alpha` resources.
+
+## What is a Pet Set?
+
+In Kubernetes, most pod management abstractions group them into disposable units of work that compose a micro service. Replication controllers for example, are designed with a weak guarantee - that there should be N replicas of a particular pod template. The pods are treated as stateless units, if one of them is unhealthy or superseded by a better version, the system just disposes it.
+
+```
+    foo.default.svc.cluster.local
+             |service|
+             /       \
+    | pod-asdf |    | pod-zxcv |
+```
+
+A Pet Set in contrast, is a group of stateful pods that form a clustered application.
+
+```
+   *.foo.default.svc.cluster.local
+    | mysql-0 | <-> | mysql-1 |
+      [pd 0]          [pd 1]
+```
+
+The co-ordinated deployment of clustered applications is notoriously hard. They require stronger notions of identity and membership, which they use in opaque internal protocols, and are especially prone to race conditions and deadlock. Traditionally administrators have deployed these applications by leveraging nodes as stable, long-lived entities with persistent storage and static ips.
+
+The goal of Pet Set is to decouple this dependency by assigning identities to individual instances of an application that are not anchored to the underlying physical infrastructure. For the rest of this document we will refer to these entities as "Pets". Our use of this term is predated by the "Pets vs Cattle" analogy.
+
+## Alpha limitations
+
+Before you start deploying applications as Pet Sets, there are a few limitations you should understand.
+
+* Pet Set is an *alpha* resource, not available in any Kubernetes release prior to 1.3.
+* As with all alpha/beta resources, it can be disable through the `--runtime-config` option passed to the apiserver, and in fact most likely will be disabled on hosted offerings of Kubernetes.
+* The only updatable field on a Pet Set is `replicas`
+* The storage for a given pet must either be provisioned by a [dynamic storage provisioner](http://releases.k8s.io/{{page.githubbranch}}/examples/experimental/persistent-volume-provisioning/README.md) based on the requested `storage class`, or pre-provisioned by an admin. Note that dynamic volume provisioning is also currently in alpha.
+* Deleting the Pet Set  *will not*  delete all pets. You will either have to manually scale it down to 0 pets first, or delete the pets yourself.
```

Comment:
It is not a first class object, it's capitalized because we define it as a term in the terminology section that doesn't have the normal meaning (dog/cat).

I don't think we need a section on pets, I define in in the what is petset section.
```
The goal of Pet Set is to decouple this dependency by assigning identities to 
individual instances of an application that are not anchored to the underlying 
physical infrastructure. For the rest of this document we will refer to these 
entities as "Pets". 
```